frontend/server/src/Controllers/Problem.php

```diff
@@ -5858,12 +5865,96 @@ public static function getLibinteractiveGenForSmarty(\OmegaUp\Request $r): array
      * @return array{smartyProperties: array{payload: CollectionDetailsPayload, title: \OmegaUp\TranslationString}, entrypoint: string}
      *
      * @omegaup-request-param string $collection_type
+     *
+     * @omegaup-request-param null|string $difficulty_range
+     * @omegaup-request-param mixed $language
+     * @omegaup-request-param int|null $max_difficulty
+     * @omegaup-request-param int|null $min_difficulty
+     * @omegaup-request-param int|null $min_visibility
+     * @omegaup-request-param mixed $offset
+     * @omegaup-request-param mixed $only_karel
+     * @omegaup-request-param mixed $order_by
+     * @omegaup-request-param mixed $page
+     * @omegaup-request-param null|string $programming_languages
+     * @omegaup-request-param null|string $query
+     * @omegaup-request-param mixed $require_all_tags
+     * @omegaup-request-param mixed $rowcount
+     * @omegaup-request-param mixed $some_tags
+     * @omegaup-request-param mixed $sort_order
      */
     public static function getCollectionsDetailsForSmarty(\OmegaUp\Request $r): array {
         $collectionType = $r->ensureString('collection_type');
 
         $collection = [];
+        $problems = [];
 
+        // Problem list
+        $offset = null;
+        $pageSize = \OmegaUp\Controllers\Problem::PAGE_SIZE;
+
+        if (is_null($r['page'])) {
+            $offset = is_null($r['offset']) ? 0 : intval($r['offset']);
+        }
+        if (!is_null($r['rowcount'])) {
+            $pageSize = intval($r['rowcount']);
+        }
+
+        [
+            'sortOrder' => $sortOrder,
+            'page' => $page,
+            'orderBy' => $orderBy,
+            'language' => $language,
+            'tags' => $tags,
+            'keyword' => $keyword,
+            'requireAllTags' => $requireAllTags,
+            'programmingLanguages' => $programmingLanguages,
+            'difficultyRange' => $difficultyRange,
+            'minVisibility' => $minVisibility,
+        ] = self::validateListParams($r);
+
+        $response = self::getList(
+            $page ?: 1,
+            $language ?: 'all',
+            $orderBy ?: 'problem_id',
+            $sortOrder ?: 'desc',
+            $offset,
+            $pageSize,
+            $tags,
+            $keyword,
+            $requireAllTags,
+            $programmingLanguages,
+            $minVisibility,
+            $difficultyRange,
+            $r->identity,
+            $r->user,
+            true
+        );
+
+        $params = [
+            'query' => $keyword,
+            'language' => $language,
+            'order_by' => $orderBy,
+            'sort_order' => $sortOrder,
+            'tag' => $tags
+        ];
+
+        $pagerItems = \OmegaUp\Pager::paginateWithUrl(
+            $response['total'],
+            $pageSize,
+            $page ?: 1,
+            '/problem/collection/([a-zA-Z0-9_+-]+)',
```

Comment:
No entiendo del todo cómo funciona esta parte.

Reply:
si te vas a https://github.com/omegaup/omegaup/blob/7b02130c308ce48e523d2a3f6df905d33d0a26ee/frontend/server/src/Pager.php#L125-L152, ahí no está haciendo nada especial con el URL que le pasas y lo está poniendo tal-cual en el enlace. entonces el enlace se mostraría como

```html
<a href="/problem/collection/([a-zA-Z0-9_+-]+)?page=0">...</a>
```

lo cual llevaría a un 404. entonces lo que hay que hacer es cambiar la parte regexosa por el parámetro que se asignó en nginx https://github.com/omegaup/omegaup/blob/7b02130c308ce48e523d2a3f6df905d33d0a26ee/frontend/server/nginx.rewrites#L78

```php
$collectionType = $r->ensureString('collection_type');
```

```suggestion
            "/problem/collection/{$collectionType}/",
```

(nota el `/` de al final, esos nunca hay que olvidarlos!)